src/SpectatorView.Unity/Assets/SpectatorView/Scripts/SpatialAlignment/MarkerDetectorCoordinateService.cs

```diff
@@ -1,190 +1,190 @@
-            DebugLog("Created artificial coordinate at origin for debugging in the editor");
-            await Task.Delay(TimeSpan.FromSeconds(1), cancellationToken).IgnoreCancellation();
-            OnNewCoordinate(coordinate.Id, coordinate);
-#else
-            DebugLog("Starting detection");
-            markerDetector.StartDetecting();
-            try
-            {
-                if (idsToLocate != null && idsToLocate.Length > 0)
-                {
-                    while (!cancellationToken.IsCancellationRequested)
-                    {
-                        bool allFound = true;
-                        foreach (int id in idsToLocate)
-                        {
-                            if (!knownCoordinates.TryGetValue(id, out ISpatialCoordinate coordinate) || coordinate.State == LocatedState.Unresolved)
-                            {
-                                DebugLog($"Marker with id: {id} wasn't found yet.");
-                                allFound = false;
-                                break;
-                            }
-                        }
-
-                        if (allFound)
-                        {
-                            DebugLog("Found all markers");
-                            break;
-                        }
-
-                        DebugLog("Waiting another second");
-                        await Task.Delay(TimeSpan.FromSeconds(1), cancellationToken).IgnoreCancellation();
-                    }
-
-                    if (cancellationToken.IsCancellationRequested)
-                    {
-                        DebugLog("We have been told to stop");
-                    }
-                }
-                else
-                {
-                    DebugLog("Just waiting to be told to stop.");
-                    await Task.Delay(-1, cancellationToken).IgnoreCancellation();
-                    DebugLog("Told to stop.");
-                }
-            }
-            finally
-            {
-                markerDetector.StopDetecting();
-                DebugLog("Stopped detection");
-            }
-#endif
-        }
-    }
-}
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Experimental.SpatialAlignment.Common;
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.MarkerDetection;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView
+{
+    /// <summary>
+    /// A marker detection based implementation of <see cref="ISpatialCoordinateService"/>.
+    /// </summary>
+    public class MarkerDetectorCoordinateService : SpatialCoordinateServiceBase<int>
```

Comment:
>MarkerDetectorCoordinateService [](start = 17, length = 31)

nit: additional subfolders may help here for clarity. Some potential examples:

SpatialAlignment/MarkerVisual
SpatialAlignment/PhysicalMarker
SpatialAlignment/Persistence